extensions/wikia/ArticleVideo/ArticleVideoContext.class.php

```diff
@@ -11,14 +11,37 @@ class ArticleVideoContext {
 	public static function isFeaturedVideoEmbedded( $title ) {
 		$wg = F::app()->wg;
 
-		return $wg->enableArticleFeaturedVideo &&
-			isset( $wg->articleVideoFeaturedVideos[$title] ) &&
-			self::isFeaturedVideosValid( $wg->articleVideoFeaturedVideos[$title] ) &&
+		if (!$wg->enableArticleFeaturedVideo) {
+			return false;
+		}
+
+		$featuredVideos = self::getFeaturedVideos();
+
+		return isset( $featuredVideos[$title] ) &&
+			self::isFeaturedVideosValid( $featuredVideos[$title] ) &&
 			// Prevents to show video on ?action=history etc.
 			!WikiaPageType::isActionPage();
 	}
 
 	/**
+	 * We are temporarily using two variables for storing the videos data
+	 * as we've run out of memory for one WF field. This will be replaced
+	 * soon by introducing a service to handle featured videos
+	 *
+	 * @return array
+	 */
+	public static function getFeaturedVideos() {
+		$wg = F::app()->wg;
+
+		$featuredVideos = array_merge(
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$featuredVideos". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)